src/util/Backtrace.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "util/Backtrace.h"
+
+#ifdef _WIN32
+
+// Backtrace-generation is not suported on windows.
+std::vector<std::string>
+getCurrentBacktrace()
+{
+    std::vector<std::string> res;
+    return res;
+}
+
+#else
+
+#define UNW_LOCAL_ONLY
+#include <cxxabi.h>
+#include <libunwind.h>
+
+std::vector<std::string>
+getCurrentBacktrace()
+{
+    std::vector<std::string> res;
+
+    unw_context_t ctxt;
```

Comment:
unw_backtrace just gets you the frame addresses; resolving them to names and demangling them requires more. the demangling function in particular expects to be given a malloc'ed buffer that it can realloc if it wants, and that the caller will free. I don't think we can really make this robust in the face of allocator corruption -- this is only going to work at all when we can at least allocate and free.

Reply:
@graydon I agree that we can't make this 100% robust, but we could make it more robust if we wanted to. We could `malloc` during program startup a small but respectable buffer (eg. 8kb) that we use for demangling. Of course if something has a ridiculously long name, this would still fail. But it would work in the general case even if we ran out of memory.

Now, with that said, I doubt it's worth the effort to actually do this. This backtrace facility is strictly better than where we were before, even if it fails when we run out of memory. And we'll still get mangled names in that case.

Reply:
Fixed.